reactor-core/src/main/java/reactor/core/publisher/FluxFlattenIterable.java

```diff
@@ -679,16 +721,18 @@ public R poll() {
 						continue;
 					}
 					current = it;
+					currentKnownToBeFinite = itFinite;
 				}
 				else if (!it.hasNext()) {
 					it = null;
+					itFinite = false;
```

Comment:
Likewise setting this here seems redundant since `itFinite` will be set again before its used.